stacks-signer/src/signer.rs

```diff
@@ -361,34 +379,48 @@ impl Signer {
+        let mut block_info = match block_validate_response {
             BlockValidateResponse::Ok(block_validate_ok) => {
                 let signer_signature_hash = block_validate_ok.signer_signature_hash;
                 // For mutability reasons, we need to take the block_info out of the map and add it back after processing
-                let Some(mut block_info) = self.blocks.remove(&signer_signature_hash) else {
-                    // We have not seen this block before. Why are we getting a response for it?
-                    debug!("Signer #{}: Received a block validate response for a block we have not seen before. Ignoring...", self.signer_id);
-                    return;
+                let mut block_info = match self.signer_db.block_lookup(&signer_signature_hash) {
+                    Ok(Some(block_info)) => block_info,
+                    Ok(None) => {
+                        // We have not seen this block before. Why are we getting a response for it?
+                        debug!("Received a block validate response for a block we have not seen before. Ignoring...");
+                        return;
+                    }
+                    Err(e) => {
+                        error!("Failed to lookup block in signer db: {:?}", e);
+                        return;
+                    }
                 };
                 let is_valid = self.verify_block_transactions(stacks_client, &block_info.block);
                 block_info.valid = Some(is_valid);
+                self.signer_db
+                    .insert_block(&block_info)
+                    .expect("Failed to insert block in DB");
                 info!(
                     "Signer #{}: Treating block validation for block {} as valid: {:?}",
                     self.signer_id,
                     &block_info.block.block_id(),
                     block_info.valid
                 );
-                // Add the block info back to the map
-                self.blocks
-                    .entry(signer_signature_hash)
-                    .or_insert(block_info)
+                block_info
             }
             BlockValidateResponse::Reject(block_validate_reject) => {
                 let signer_signature_hash = block_validate_reject.signer_signature_hash;
-                let Some(block_info) = self.blocks.get_mut(&signer_signature_hash) else {
-                    // We have not seen this block before. Why are we getting a response for it?
-                    debug!("Signer #{}: Received a block validate response for a block we have not seen before. Ignoring...", self.signer_id);
-                    return;
+                let mut block_info = match self.signer_db.block_lookup(&signer_signature_hash) {
+                    Ok(Some(block_info)) => block_info,
+                    Ok(None) => {
+                        // We have not seen this block before. Why are we getting a response for it?
```

Comment:
I think actually all the logs accidentally remove "Signer #{}" from them. Make sure to add these back.